diplomatic/src/rocket/FPU.scala

```diff
@@ -14,13 +13,144 @@ import org.chipsalliance.rocket.Instructions._
 import freechips.rocketchip.util._
 import freechips.rocketchip.util.property
 
-case class FPUParams(
-  minFLen: Int = 32,
-  fLen: Int = 64,
-  divSqrt: Boolean = true,
-  sfmaLatency: Int = 3,
-  dfmaLatency: Int = 4
-)
+trait FPUParams {
```

Comment:
Why moving this around

Reply:
In the original code, parameters and helper functions are scatter around in `FPUParams` and `HasFPUParameters` (the latter trait actually does way more things than what its name implies). There was an abstract class used to merged inherited parameters from outside and `HasFPUParameters`. At first I tried to address the problem using trait; but since `fLen` cannot be determined and should be a kind of parameter, and the helper functions need `fLen` nearly everywhere, later on I refactor the parameters and helper functions into a new abstract class `FPUModule` for parameters to pass through. After all, it can be better by refactoring parameters not dependent to `fLen` into another trait, and I am still exploring this option.